exercises/hexadecimal/hexadecimal_test.go

```diff
@@ -42,18 +43,33 @@ var testCases = []struct {
 
 func TestParseHex(t *testing.T) {
 	for _, test := range testCases {
-		switch out, err := ParseHex(test.in); {
-		case err != nil:
-			if test.errCase == "none" {
+		out, err := ParseHex(test.in)
+		if test.errCase != "none" {
+			// check if err is of error type
+			var _ error = err
+
+			// we expect error
+			if err == nil {
+				t.Errorf("ParseHex(%q): expected an error, but error is nil",
+					test.in)
+			}
+
+			if !strings.Contains(strings.ToLower(err.Error()), test.errCase) {
+				t.Errorf(
+					"ParseHex(%q) returned error %q. Expected error %s.",
+					err, test.errCase)
```

Comment:
the format string has three vars (`%q`, `%q`, `%s`) but I only see two vars here (`err, `test.errCase`) - can it be made consistent? Probably needs an `test.in`.